tests/postgres/alltypes_test.go

```diff
@@ -931,6 +932,43 @@ func TestTimeExpression(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestIntervalUpsert(t *testing.T) {
+	testutils.ExecuteInTxAndRollback(t, db, func(tx *sql.Tx) {
+		stmt := SELECT(Employee.AllColumns).FROM(Employee).
+			WHERE(Employee.EmployeeID.EQ(Int(1)))
+
+		//Validate initial dataset
+		var windy model.Employee
+		err := stmt.Query(tx, &windy)
+		assert.Equal(t, windy.EmployeeID, int32(1))
+		assert.Equal(t, windy.FirstName, "Windy")
+		assert.Equal(t, windy.LastName, "Hays")
+		assert.Equal(t, *windy.PtoAccrual, "22:00:00")
+		assert.Nil(t, err)
+		windy.PtoAccrual = ptr.Of("3h")
+		//Update data
+		updateStmt := Employee.UPDATE(Employee.PtoAccrual).SET(
+			Employee.PtoAccrual.SET(INTERVAL(3, HOUR)),
+		).WHERE(Employee.EmployeeID.EQ(Int(1))).RETURNING(Employee.AllColumns)
+
+		err = updateStmt.Query(tx, &windy)
+		err = stmt.Query(tx, &windy)
+		assert.Nil(t, err)
+		assert.Equal(t, *windy.PtoAccrual, "03:00:00")
+		//Upsert dataset with a different value
+		windy.PtoAccrual = ptr.Of("5h")
+		insertStmt := Employee.INSERT(Employee.AllColumns).
+			MODEL(&windy).
+			ON_CONFLICT(Employee.EmployeeID).
+			DO_UPDATE(SET(
+				Employee.PtoAccrual.SET(Employee.EXCLUDED.PtoAccrual),
+			)).RETURNING(Employee.AllColumns)
+		err = insertStmt.Query(tx, &windy)
+		assert.Nil(t, err)
+		assert.Equal(t, *windy.PtoAccrual, "05:00:00")
```

Comment:
We also need to make sure `SET' appears in the generated sql. Check `testutils.AssertStatementSql`.  
L937-L948 can be removed, since is unrelated to this test and there plenty of select scan tests. 
update and upsert are independent test case and can be grouped into separate sub tests `t.Run("update",...` and `t.Run("upsert"`.

Reply:
done